scipy/integrate/_ivp/ivp.py

```diff
@@ -660,8 +660,8 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
         ts = np.array(ts)
         ys = np.vstack(ys).T
     elif ts:
-        ts = np.hstack(ts)
-        ys = np.hstack(ys)
+        ts = np.hstack(ts) if ts else np.empty(0)
```

Comment:
This should not be needed, because `if ts` is always true - it's in an `elif ts` block.

And `ts` and `ys` should always be the same size, otherwise something is wrong.

Reply:
Ah. Seems I copied a fix over from my older version of scipy without carefully checking. The bug I'm referencing was already fixed in #14755 , which changes the `else` in the previous line to `elif ts`

Returning zero-size `np.empty` arrays seems more correct than empty lists to me though, but this is extremely minor.